src/core/surface/completion_queue.c

```diff
@@ -184,11 +203,34 @@ grpc_event grpc_completion_queue_next(grpc_completion_queue *cc,
   return ret;
 }
 
+static void add_plucker(grpc_completion_queue *cc, void *tag,
+                        grpc_pollset_worker *worker) {
+  GPR_ASSERT(cc->num_pluckers != MAX_PLUCKERS);
```

Comment:
As a followup to my previous note, should this be some kind of resource exhaustion error code rather than an assertion? Is it inconceivable that this will trip?